core/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -222,7 +223,8 @@ public ThreadPool(Settings settings) {
         int halfProcMaxAt5 = Math.min(((availableProcessors + 1) / 2), 5);
         int halfProcMaxAt10 = Math.min(((availableProcessors + 1) / 2), 10);
         Map<String, Settings> defaultExecutorTypeSettings = new HashMap<>();
-        add(defaultExecutorTypeSettings, new ExecutorSettingsBuilder(Names.GENERIC).size(4 * availableProcessors).keepAlive("30s"));
+        int genericThreadPoolSize = Math.min(512, Math.max(128, 4 * availableProcessors));
```

Comment:
can we document as why are we doing this? I'm also wondering if we should just keep things super simple and a fixed default of 512 . I had to read this line 3 times to make sure I get what it means.